airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_tracker/JobTracker.java

```diff
@@ -168,30 +179,130 @@ public void trackSync(final Job job, final JobState jobState) {
+   * array) then returns a map of {null: toMetadataValue(config)}.
+   */
+  private static Map<String, Object> configToMetadata(final JsonNode config, final JsonNode schema) {
+    if (schema.hasNonNull("const")) {
+      // If this schema is a const, then just dump it into a map:
+      // * If it's an object, flatten it
+      // * Otherwise, do some basic conversions to value-ish data.
+      // It would be a weird thing to declare const: null, but in that case we don't want to report null
+      // anyway, so explicitly use hasNonNull.
+      return flatten(config);
+    } else if (schema.has("oneOf")) {
+      // If this schema is a oneOf, then find the first sub-schema which the config matches
+      // and use that sub-schema to convert the config to a map
+      final JsonSchemaValidator validator = new JsonSchemaValidator();
+      for (final Iterator<JsonNode> it = schema.get("oneOf").elements(); it.hasNext();) {
+        final JsonNode subSchema = it.next();
+        if (validator.test(subSchema, config)) {
+          return configToMetadata(config, subSchema);
+        }
+      }
+      // If we didn't match any of the subschemas, then something is wrong. Bail out silently.
+      return emptyMap();
+    } else if (config.isObject()) {
+      // If the schema is not a oneOf, but the config is an object (i.e. the schema has "type": "object")
+      // then we need to recursively convert each field of the object to a map.
+      // Note: this doesn't handle additionalProperties, so if a config contains properties that are not
+      // explicitly declared in the schema, they will be ignored.
+      final Map<String, Object> output = new HashMap<>();
+      final JsonNode maybeProperties = schema.get("properties");
+      if (maybeProperties != null) {
+        for (final Iterator<Entry<String, JsonNode>> it = config.fields(); it.hasNext();) {
+          final Entry<String, JsonNode> entry = it.next();
+          final String field = entry.getKey();
+          final JsonNode value = entry.getValue();
+          if (maybeProperties.hasNonNull(field)) {
+            mergeMaps(output, field, configToMetadata(value, maybeProperties.get(field)));
           }
         }
       }
+      return output;
+    } else if (config.isBoolean()) {
+      return singletonMap(null, config.asBoolean());
+    } else if ((!config.isTextual() && !config.isNull()) || (config.isTextual() && !config.asText().isEmpty())) {
+      // This is either non-textual (e.g. integer, array, etc) or non-empty text
+      return singletonMap(null, SET);
+    } else {
+      // Otherwise, this is an empty string, so just ignore it
+      return emptyMap();
     }
+  }
 
-    return output;
+  /**
+   * Naively flattens a JsonNode, or dumps it into a {null: node} map. Does _not_ mask values; ONLY
+   * use this if you're sure that node does NOT contain potentially-sensitive data.
+   */
+  private static Map<String, Object> flatten(final JsonNode node) {
```

Comment:
is it generic enough that it could be moved into `airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java`?